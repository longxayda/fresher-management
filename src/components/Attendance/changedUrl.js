function setTime() {
    var today = new Date();
    var time = today.getHours();
    var text = ""
    var tmp = text
    // if (time == 7 || time == 11 || time == 17 || time == 18 || time == 20){
        var possible = "abcdefghijklmnopqrstuvwxyz123456789";
        for (var i = 0; i < 5; i++)
            text += possible.charAt(Math.floor(Math.random() * possible.length));
        console.log(text)
    // }
    return text;

}

export default setTime;