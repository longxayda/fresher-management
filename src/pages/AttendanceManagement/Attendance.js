import React from "react";
import {
  Button,
  Container,
  Row,
  Col,
  Form
} from "react-bootstrap";
import { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { postAttendance } from "redux/attendanceManagementSlice/attendanceSlice";
import { attendMessageSeletor } from "redux/selectors/attendanceSelectors/adminTicketSelectors"

function Attendance() {
  var today = new Date();
  var date = today.getDate()+'-'+(today.getMonth()+1)+'-'+today.getFullYear();
  var time = today.getHours() + ":" + today.getMinutes();
  var dateTime = date+' '+time;

  const [typePassword, setTypePassword] = useState("password");
  const changeTypePassword = () => {
      if(typePassword == "password"){
          setTypePassword("text");
      }
      else {
          setTypePassword("password")
      }
  };

  const dispatch = useDispatch();
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  let errorMsg = useSelector(attendMessageSeletor);
  let className = ""
  if (errorMsg != null){
    className = "text-danger"
  }
  else {
    className = "text-primary"
  }

  const handleSubmit = async (event) => {
    event.preventDefault()
    await dispatch(postAttendance({
      "username" : username,
      "password" : password
    }));
  }


  return (
    <>
      <Container className="attendance-form-sreen">
        <Row className="justify-content-center">
          <Col className="attend-form" md="6">
            <h2>Attendance FSOFT</h2>

            <Row className="justify-content-between">              
              <Col md="3"><h4>Time</h4></Col>
              <Col><h4>{dateTime}</h4></Col>
            </Row>

            <Form style={{fontSize: "20px"}}>
              <Form.Group className="md-3">
                <Row style={{margin: "20px -10px 0"}}>
                  <Form.Label htmlFor="username" column md="3">
                    Username
                  </Form.Label>
                  <Col md="9">
                    <Form.Control
                      id="username"
                      type="text"
                      value = {username}
                      onChange = {e => setUsername(e.target.value)}
                    />
                  </Col>
                </Row>

                <Row style={{margin: "20px -10px 0"}}>
                  <Form.Label htmlFor="username" column md="3" >
                    Password
                  </Form.Label>
                  <Col md="9">
                    <Form.Control
                      id="password"
                      type = {typePassword}
                      value = {password}
                      onChange = {e => setPassword(e.target.value)}
                      placeholder = "Password"
                    />
                    <Form.Text className={className}>
                      {errorMsg}
                    </Form.Text>
                  </Col>
                </Row>
                <Row style={{margin: "20px -10px 0"}}>
                  <Col md="3"/>
                  <Col md="9">
                    <Form.Check.Label>
                      <Row>
                        <Col md="1">
                          <Form.Check.Input type="checkbox" onChange={changeTypePassword}/>
                        </Col>
                        <Col md="10">
                          <h5 style={{margin: "5px"}}>Show password</h5>
                        </Col>
                      </Row>
                    </Form.Check.Label>
                  </Col>
                </Row>
              </Form.Group>
            </Form>

            <Button
              id="attend"
              className="btn-fill login-form-button"
              style={{float: "right", margin: "30px 0 0"}}
              onClick={ handleSubmit }
            >
              Attend
            </Button>      
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default Attendance;