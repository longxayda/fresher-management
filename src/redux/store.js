import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
