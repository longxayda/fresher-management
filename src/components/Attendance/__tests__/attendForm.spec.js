import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });
import store from 'redux/store';
import { Provider } from "react-redux";
import Attendance from 'pages/AttendanceManagement/Attendance.js'

describe('Test Attendance Form Sreen', () => {
    let component;
    beforeEach(() => {
        component = mount(<Provider store = { store }> <Attendance /> </Provider>);
    });

    test('has the correct class', () => {
        expect(component.find('.attendance-form-sreen').exists()).toEqual(true);
    });

    test('render Attendance Form Sreen\'s component correctly', () => {
        component = mount(<Provider store = { store }> <Attendance /> </Provider>);
        expect(component.html()).toMatchSnapshot();
    });
})
