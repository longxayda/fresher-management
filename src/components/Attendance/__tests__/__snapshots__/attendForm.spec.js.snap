// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Attendance Form Sreen render Attendance Form Sreen's component correctly 1`] = `" <div class=\\"attendance-form-sreen container\\"><div class=\\"justify-content-center row\\"><div class=\\"attend-form col-md-6\\"><h2>Attendance FSOFT</h2><div class=\\"justify-content-between row\\"><div class=\\"col-md-3\\"><h4>Time</h4></div><div class=\\"col\\"><h4>13-8-2022 12:3</h4></div></div><form style=\\"font-size: 20px;\\" class=\\"\\"><div class=\\"md-3\\"><div style=\\"margin: 20px -10px 0px;\\" class=\\"row\\"><label for=\\"username\\" class=\\"form-label col-form-label col-md-3\\">Username</label><div class=\\"col-md-9\\"><input type=\\"text\\" id=\\"username\\" class=\\"form-control\\" value=\\"\\"></div></div><div style=\\"margin: 20px -10px 0px;\\" class=\\"row\\"><label for=\\"username\\" class=\\"form-label col-form-label col-md-3\\">Password</label><div class=\\"col-md-9\\"><input placeholder=\\"Password\\" type=\\"password\\" id=\\"password\\" class=\\"form-control\\" value=\\"\\"><small class=\\"text-danger form-text\\"></small></div></div><div style=\\"margin: 20px -10px 0px;\\" class=\\"row\\"><div class=\\"col-md-3\\"></div><div class=\\"col-md-9\\"><label class=\\"form-check-label\\"><div class=\\"row\\"><div class=\\"col-md-1\\"><input type=\\"checkbox\\" class=\\"form-check-input\\"></div><div class=\\"col-md-10\\"><h5 style=\\"margin: 5px;\\">Show password</h5></div></div></label></div></div></div></form><button type=\\"button\\" id=\\"attend\\" style=\\"float: right; margin: 30px 0px 0px;\\" class=\\"btn-fill login-form-button btn btn-primary\\">Attend</button></div></div></div> "`;
